File: space-game/Core.Tests/Classes/MassTests.vb
Imports Core.Classes
Imports Core.Tests.Data
Imports NUnit.Framework

Namespace Classes
    <TestFixture>
    Public Class MassTests

        <TestCase(0.0)>
        <TestCase(56.0)>
        <TestCase(0.00000001)>
        <TestCase(25444588655.0)>
        Public Sub FromKilograms_WhenCalled_StoresCorrectValue(kilograms As Double)

            Dim m = Mass.FromKilograms(kilograms)

            Dim result = m.Kilograms

            Assert.AreEqual(kilograms, result)
        End Sub

        <TestCase(0.0)>
        <TestCase(56.0)>
        <TestCase(0.00000001)>
        <TestCase(25444588655.0)>
        Public Sub FromSolarMasses_WhenCalled_StoresCorrectValue(solarMasses As Double)
            Dim acceptableDelta = Mass.FromKilograms(1).Kilograms
            Dim m = Mass.FromSolarMasses(solarMasses)

            Dim result = m.SolarMasses

            Assert.AreEqual(solarMasses, result, acceptableDelta)
        End Sub

        <TestCase(0.0)>
        <TestCase(56.0)>
        <TestCase(0.00000001)>
        <TestCase(25444588655.0)>
        Public Sub FromEarthMasses_WhenCalled_StoresCorrectValue(earthMasses As Double)
            Dim acceptableDelta = Mass.FromKilograms(1).Kilograms
            Dim m = Mass.FromEarthMasses(earthMasses)

            Dim result = m.EarthMasses

            Assert.AreEqual(earthMasses, result, acceptableDelta)
        End Sub

        <TestCaseSource(GetType(MassTestsData), NameOf(MassTestsData.Equals_WhenComparing_ReturnsExpected_Data))>
        Public Sub Equals_WhenComparing_ReturnsExpected(this As Mass, that As Mass, expected As Boolean)
            Dim result = this.Equals(that)

            Assert.AreEqual(expected, result)
        End Sub

    End Class
End Namespace

File: space-game/Core.Tests/Classes/MassTests.cs
using Core.Classes;
using Core.Tests.Data;
using NUnit.Framework;

namespace Core.Tests.Classes
{
    [TestFixture]
    public class MassTests
    {
        [TestCase(0.0)]
        [TestCase(56.0)]
        [TestCase(1E-08)]
        [TestCase(25444588655.0)]
        public void FromKilograms_WhenCalled_StoresCorrectValue(double kilograms)
        {
            Mass m = Mass.FromKilograms(kilograms);

            double result = m.Kilograms;

            Assert.AreEqual(kilograms, result);
        }

        [TestCase(0.0)]
        [TestCase(56.0)]
        [TestCase(1E-08)]
        [TestCase(25444588655.0)]
        public void FromSolarMasses_WhenCalled_StoresCorrectValue(double solarMasses)
        {
            double acceptableDelta = Mass.FromKilograms(1).Kilograms;
            Mass m = Mass.FromSolarMasses(solarMasses);

            double result = m.SolarMasses;

            Assert.AreEqual(solarMasses, result, acceptableDelta);
        }

        [TestCase(0.0)]
        [TestCase(56.0)]
        [TestCase(1E-08)]
        [TestCase(25444588655.0)]
        public void FromEarthMasses_WhenCalled_StoresCorrectValue(double earthMasses)
        {
            double acceptableDelta = Mass.FromKilograms(1).Kilograms;
            Mass m = Mass.FromEarthMasses(earthMasses);

            double result = m.EarthMasses;

            Assert.AreEqual(earthMasses, result, acceptableDelta);
        }

        [TestCaseSource(typeof(MassTestsData), nameof(MassTestsData.Equals_WhenComparing_ReturnsExpected_Data))]
        public void Equals_WhenComparing_ReturnsExpected(Mass @this, Mass that, bool expected)
        {
            bool result = @this.Equals(that);

            Assert.AreEqual(expected, result);
        }
    }
}
